src/pipeline.cc

```diff
@@ -693,10 +694,15 @@ class PipelineWorker : public AsyncWorker {
           // the overlayGravity was used for extract_area, therefore set it back to its default value of 0
           baton->overlayGravity = 0;
         }
-        // Ensure overlay is premultiplied sRGB
-        overlayImage = overlayImage.colourspace(VIPS_INTERPRETATION_sRGB).premultiply();
-        // Composite images with given gravity
-        image = Composite(overlayImage, image, baton->overlayGravity);
+        if(baton->overlayCutout) {
+          // 'cut out' the image, premultiplication is not required
```

Comment:
The image to be overlaid has been pre-multiplied, so we'll either need to also pre-multiply here or not pre-multiply the image to be overlaid via the logic that calculates `shouldPremultiplyAlpha`.